### ISSUE TITLE
Cancel jump to definition on paste before mouseup

### ISSUE THREAD

**Issue by rkusa** [zed-industries/zed#5209]
### Check for existing issues

- [X] Completed

### Describe the bug

I often accidentally jump to a definition when instead I am just wanting to place the cursor and paste. While this is mostly due to bad timing between my left and right hand, I never noticed that in other editors. After digging into it, I can confirm that VSCode and Sublime cancel the jump to definition when text is pasted before the mouseup event is fired.

### To reproduce

1. Copy some code
2. Keep <kbd>⌘</kbd> pressed (for the duration of all the following steps)
3. Mousedown somewhere where a `Jump to Definition` is possible (just down, don't release yet)
4. Paste the code (<kbd>⌘</kbd> + <kbd>v</kbd>)
5. Release the mouse (mouseup)
6. Zed will jump to the definition

So basically the one hand wants to press <kbd>⌘</kbd> + <kbd>v</kbd>, after the other hand clicked at the location where you wanna paste. But instead of mouse down -> mouse release -> <kbd>⌘</kbd> -> <kbd>v</kbd>, you easily end up with <kbd>⌘</kbd> -> mouse down -> <kbd>v</kbd> -> mouse release which causes the unintended jump to definition.

### Expected behavior

Zed is not jumping to definition. The jump to definition is canceled by pasting the code (or changing the buffer in general maybe).

### Environment

Zed 0.55.0 – /Applications/Zed.app \nmacOS 12.6 \narchitecture arm64

**Comment by rkusa** [comment]
I am still able to reproduce this issue in the latest version of Zed 👍
